addon/core/rdfa-processor.ts

```diff
@@ -125,63 +184,112 @@ export function preprocessRDFa(dom: Node) {
       const contentId = ensureId(contentNode as HTMLElement);
+      if (quad.object.termType !== 'Literal') {
+        throw new Error(
+          'unexpected quad object type for quad referring to literal node',
+        );
+      }
+      const datatype: Omit<RDF.NamedNode, 'equals'> | undefined =
+        quad.object.datatype && quad.object.datatype.value !== LANG_STRING
+          ? { termType: 'NamedNode', value: quad.object.datatype.value }
+          : undefined;
+      console.log('datatype', datatype);
       result.push({
-        type: 'external',
         predicate: quad.predicate.value,
         object: {
-          type: 'literal',
+          termType: 'LiteralNode',
           rdfaId: contentId,
+          datatype: datatype ?? { termType: 'NamedNode', value: '' },
+          language: datatype ? '' : quad.object.language ?? '',
         },
       });
     }
     return result;
   } else {
     // check if this quad refers to a resourceNode
-    if (
-      quad.object.termType === 'BlankNode' ||
-      quad.object.termType === 'NamedNode'
-    ) {
+    const { object } = quad;
+    if (object.termType === 'BlankNode' || object.termType === 'NamedNode') {
       const resourceNode = datastore
         .getResourceNodeMap()
-        .getFirstValue({ subject: quad.object });
+        .getFirstValue({ subject: object });
       if (resourceNode) {
         return [
           {
-            type: 'external',
+            predicate: quad.predicate.value,
+            object: { termType: 'ResourceNode', value: object.value },
+          },
+        ];
+      } else {
+        // this is just to make typescript happy
+        if (object.termType === 'BlankNode') {
+          return [
+            {
+              predicate: quad.predicate.value,
+              object: { value: object.value, termType: object.termType },
+            },
+          ];
+        } else {
+          return [
+            {
+              predicate: quad.predicate.value,
+              object: { value: object.value, termType: object.termType },
+            },
+          ];
+        }
+      }
+    }
+    // neither a content nor resource node, so just a plain attribute
+    if (quad.object.termType === 'Literal') {
+      const { contentDatatype, contentLanguage, contentPredicate } = entry;
+      if (
+        contentPredicate &&
+        quad.predicate.equals(contentPredicate) &&
+        (!contentDatatype || contentDatatype?.equals(quad.object.datatype)) &&
+        (!contentLanguage ||
+          contentLanguage.toUpperCase() === quad.object.language.toUpperCase())
+      ) {
+        return [
+          {
             predicate: quad.predicate.value,
             object: {
-              type: 'resource',
-              resource: quad.object.value,
+              termType: 'ContentLiteral',
+              datatype: contentDatatype ?? { termType: 'NamedNode', value: '' },
+              language: contentLanguage ?? '',
```

Comment:
This probably needs to be changed to:

```suggestion
              language: contentDatatype ?'' : contentLanguage ?? '',
```